ref(autofix): Move actions to document and remove message box

roaga opened the pull request:
(Note for issues team if you open this PR: this PR only contains changes inside the Sentry AI section of the drawer)

I know this PR has a big change count, but it's really just one change: removing the message box. The user's actions are now on the root cause and fix cards, and logs and user input are in the output stream. It's a lot of moving existing code and tests, and deleting dead code. I also had to include some styling changes for this to all look coherent and readable, plus tacking on some other styling changes for other existing readability issues. Nothing has changed on functionality.

(The insight card justifications will look better once the backend change goes in)

<img width="694" alt="Screenshot 2025-01-10 at 8 18 52 AM" src="https://github.com/user-attachments/assets/70e01bc4-f508-45e1-880e-7f94a7bde67f" />
<img width="723" alt="Screenshot 2025-01-10 at 8 19 00 AM" src="https://github.com/user-attachments/assets/a96d362d-5ac9-4e77-b60f-65c30cbd2c91" />
<img width="675" alt="Screenshot 2025-01-10 at 8 49 31 AM" src="https://github.com/user-attachments/assets/423aec23-c0e5-44fd-9396-18b97d306074" />
<img width="662" alt="Screenshot 2025-01-10 at 11 42 49 AM" src="https://github.com/user-attachments/assets/ac230aae-cc94-4146-aa15-ee79ce8605ce" />
<img width="653" alt="Screenshot 2025-01-14 at 6 50 00 AM" src="https://github.com/user-attachments/assets/4cb3449c-6656-4663-9501-9212376af753" />
<img width="634" alt="Screenshot 2025-01-14 at 6 49 49 AM" src="https://github.com/user-attachments/assets/597c94e3-4833-45ad-8b3d-559d12029161" />

roaga committed:
Move actions to document and remove message box

roaga committed:
Fix padding

roaga committed:
Fix padding

roaga committed:
Increase button size

roaga committed:
Update

roaga committed:
Switch icon back

jennmueng commented on static/app/components/events/autofix/autofixSteps.tsx:
```diff
@@ -79,15 +73,12 @@ export function Step({
-                  Autofix encountered an error.
-                  <br />
-                  Restarting step from scratch...
+                  Autofix encountered an error. Restarting step from scratch...
```
```suggestion
                  {t('Autofix encountered an error. Restarting step from scratch...')}
```

jennmueng commented on static/app/components/events/autofix/types.ts:
```diff
@@ -98,30 +98,9 @@ export type CodeSnippetContext = {
 export type AutofixInsight = {
-  breadcrumb_context: BreadcrumbContext[];
-  codebase_context: CodeSnippetContext[];
   insight: string;
   justification: string;
-  stacktrace_context: StacktraceContext[];
 };
```
are insights becoming just strings with no code context?

roaga replied:
They will have code context, but it will all come as a Markdown string instead of structured output (once my Seer-side change goes through).

jennmueng reviewed (commented):
I see this is still draft, is this ready to go?

roaga committed:
Update static/app/components/events/autofix/autofixSteps.tsx

Co-authored-by: Jenn Mueng <30991498+jennmueng@users.noreply.github.com>

roaga committed:
Update static/app/components/events/autofix/autofixOutputStream.tsx

Co-authored-by: Jenn Mueng <30991498+jennmueng@users.noreply.github.com>

roaga commented:
@jennmueng this PR is ready for review but I will likely wait to merge and instead make another PR to merge into this one for round 2 of changes.

roaga committed:
Update static/app/components/events/autofix/autofixRootCause.tsx

Co-authored-by: Jenn Mueng <30991498+jennmueng@users.noreply.github.com>

roaga committed:
feat(autofix): UX improvements (#83800)

1. Collapsible insight cards. Can be collapsed manually, and also
auto-collapses when a step is "completed" (root cause selected or
PR/branch made).
2. Readability and saving vertical space by swapping arrows for a
vertical line
3. Replace rethink on arrows with editing/adding insight cards. This is
more expected for a document.
4. Support commenting on document. Click any text on an insight card,
root cause, or diff, and ask questions or leave feedback, as expected
for a document. Autofix responds, and if it detects that action is
needed, it will automatically trigger a rethink.

This PR requires a backend change in Seer to go in first.

<img width="650" alt="Screenshot 2025-01-21 at 7 19 15 PM"
src="https://github.com/user-attachments/assets/cada2f77-fe59-438c-a3ed-cbfbc8526ec7"
/>
<img width="820" alt="Screenshot 2025-01-21 at 7 19 26 PM"
src="https://github.com/user-attachments/assets/ab4f43e6-bebb-44c2-a2fb-1b097753fd5c"
/>
<img width="806" alt="Screenshot 2025-01-21 at 7 20 24 PM"
src="https://github.com/user-attachments/assets/ed44c3d8-6b46-4595-8b64-4fcd3bd57e6f"
/>
<img width="600" alt="Screenshot 2025-01-21 at 7 26 16 PM"
src="https://github.com/user-attachments/assets/c0de04b7-26e2-4c10-bd69-f26a4067d98f"
/>
